### PR TITLE
Changing Postgres instance type from Default to Replica forces a new resource to be created

### PR DESCRIPTION
**Pull request by sheffsky**
According to the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/postgresql/concepts-read-replicas)
> The standalone server can't be made into a replica again.

So changing instance type from Default to Replica should lead to resource re-creation.

### CODE REVIEW

File: internal/services/postgres/postgresql_server_resource.go [hunk 1]
```diff
@@ -446,6 +446,15 @@ func resourcePostgreSQLServer() *pluginsdk.Resource {
 				}
 				return false
 			}),
+			pluginsdk.ForceNewIfChange("create_mode", func(ctx context.Context, old, new, meta interface{}) bool {
+				oldMode := old.(postgresql.CreateMode)
+				newMode := new.(postgresql.CreateMode)
```
Comment (tombuildsstuff) [inline internal/services/postgres/postgresql_server_resource.go:451]: these'll need to be:

```suggestion
				oldMode := postgresql.CreateMode(old.(string))
				newMode := postgresql.CreateMode(new.(string))
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @sheffsky 

Thanks for this PR - taking a look through here whilst on the whole this looks pretty good we need to cast these to strings since these are coming from the Plugin SDK as string values rather than the `postgresql.CreateMode` type - but if we can fix those up and the tests pass then we should be able to take another look.

Thanks!

**Comment by sheffsky** [comment]
Hey @tombuildsstuff 
Thanks for the review! I've pushed the proposed changes.

**Review by katbyte — APPROVED** [review]
Thanks @sheffsky - LGTM 🏗️

**Pull request merged**